beacon-chain/core/blocks/block_operations.go

```diff
@@ -0,0 +1,62 @@
+package blocks
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	v "github.com/prysmaticlabs/prysm/beacon-chain/core/validators"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+var log = logrus.WithField("prefix", "state")
+
+// ProcessProposerSlashings is one of the operations performed
+// on each processed beacon block to penalize proposers based on
+// slashing conditions if any slashable events occurred.
+func ProcessProposerSlashings(
+	validatorRegistry []*pb.ValidatorRecord,
+	proposerSlashings []*pb.ProposerSlashing,
+	currentSlot uint64,
+) ([]*pb.ValidatorRecord, error) {
+	if uint64(len(proposerSlashings)) > params.BeaconConfig().MaxProposerSlashings {
+		return nil, fmt.Errorf("number of proposer slashings exceeds threshold: %v", len(proposerSlashings))
+	}
+	// TODO(#781): Verify BLS according to the specification in the "Proposer Slashings"
+	// section of block operations.
+	for idx, slashing := range proposerSlashings {
+		if err := verifyProposerSlashing(validatorRegistry, slashing, currentSlot); err != nil {
+			return nil, fmt.Errorf("could not verify proposer slashing #%d: %v", idx, err)
+		}
+		proposer := validatorRegistry[slashing.GetProposerIndex()]
+		if proposer.Status != uint64(params.ExitedWithPenalty) {
```

Comment:
This should actually use StatusCode: https://github.com/prysmaticlabs/prysm/blob/b4f8c0afaf4c3d0de48f22c47cb30d4b80e44735/proto/beacon/p2p/v1/types.proto#L105

Reply:
Ignore this, `status_code` is now deprecated and `status` is correct. #1082